packages/cli/test/models/ContractManager.test.js

```diff
@@ -61,14 +63,16 @@ contract('ContractManager', function([_, from]) {
                 bytecode: '0x124',
                 contractName: 'Foo',
               };
-              FileSystem.writeJson(`${this.testDir}/build/contracts/Foo.json`, builtContract);
+              this.fooContractPath = `${this.testDir}/build/contracts/Foo.json`;
+              FileSystem.writeJson(this.fooContractPath, builtContract);
               this.projectFile = new ProjectFile(`${this.testDir}/zos.json`);
               this.contractManager = new ContractManager(this.projectFile);
               sinon.stub(ConfigManager, 'getBuildDir').returns(`${this.testDir}/build/contracts`);
               sinon.stub(Contracts, 'getLocalContractsDir').returns(`${this.testDir}/contracts`);
             });
 
             afterEach(function() {
+              FileSystem.remove(this.fooContractPath);
```

Comment:
Why is this needed? Is is related to `test-environment` or `mocha`?

Reply:
Not related to `test-env` or `mocha` but it was causing tests to fail because `Foo.json` would be left in the file system and subsequent Compiler tests would fail because `Foo.json` is not a real artifact. I have no idea why doesn't fail on CircleCI.

Reply:
Strange :man_shrugging: Removing it does seem like the way to go though